### PR TITLE
feat: support view-test-artifacts rather than a single artifact

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Add a utility to make it easier to look at multiple failing test artifacts in parallel.

asset-download

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
Verified locally.

#### Checklist
- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by phani-srikar** [comment]
Looks like the `lint` job in cb is failing, can you run `prettier` on the changed file to fix it?

File: scripts/view-test-artifacts.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    )
+    .filter(testArtifactIsComplete);
+};
+
+const downloadSingleTestArtifact = async (tempDir: string, artifact: Required<TestArtifact>): Promise<unknown[]> => {
+  const artifactDownloadPath = path.join(tempDir, artifact.jobName);
+  fs.mkdirSync(artifactDownloadPath);
+  const [Bucket, ...rest] = artifact.artifactLocation.split(':::')[1].split('/');
+  const Prefix = rest.join('/');
+  const listObjectsResponse = await s3.listObjects({ Bucket, Prefix }).promise();
```
Comment (phani-srikar) [inline scripts/view-test-artifacts.ts:109]: As per the AppSec guidance, we need to enable versioning on the buckets which would need some additional filtering here to only get the latest versions of objects.

Comment (alharris-at) [reply scripts/view-test-artifacts.ts:109]: Sounds good, I'll wait til that change is out and we've had some runs to verify behavior before we move forward w/ that.

Comment (phani-srikar) [reply scripts/view-test-artifacts.ts:106]: I've tested this script with versioned buckets and it works. Good to go.

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**